### ISSUE TITLE
time series indexer

### ISSUE THREAD

**Issue by flennerhag** [flennerhag/mlens#25]
A K-fold like time series indexer for time series cross-validation. The indexer should do non-overlapping: 

| fold |  train obs  |  test obs  |
| ---- | ---------- | --------- | 
|   0    |        0        |     1          |
|   1     |      0, 1      |      2        |
|   2     |    0, 1, 2   |      3         |

**Comment by JoshuaC3** [comment]
This would be great.

In addition, it would be useful to add in a base training size e.g. 3 (of the 6 "folds"). This better mimics many real life scenarios where you have a good amount of training data with an expanding window.

fold | train obs | test obs
-- | -- | --
0 | 0, 1, 2, 3 | 4
1 | 0, 1, 2, 3, 4 | 5
2 | 0, 1, 2, 3, 4, 5 | 6

It would also be useful to include a moving window size, e.g. 4 (on 6 folds)

fold | train obs | test obs
-- | -- | --
0 | 0, 1, 2, 3 | 4
1 | 1, 2, 3, 4 | 5
2 | 2, 3, 4, 5 | 6

Ultimately, this could all be achieved by allowing a user to define their own indices for the train-test-split.

**Comment by flennerhag** [comment]
Agree. A ``make_indexer`` function would be a great utility to convert any list of train-test splits of fold generator into a mlens compatible indexer. Actually think the ``mlens.index`` code base (in ``0.2.0``) already has the base code needed for that, worth looking into! 

Come think of it, the time series indexer is quite straightforward to implement as the folds are contiguous: 

```python
# Params
require: burn_in
require: window_size
require: test_size

start_index = 0
stop_index = burn_in
stop = False
while not stop:
    start_index += window_size  # if moving window
    stop_index += window_size

    if stop_index + test_size >= X.shape[0]:
        stop = True
        # Something here to grab the last obs in the final fold

    train_index = (start_index, stop_index)
    test_index = (stop_index, stop_index + test_size)  # test_size might not always be 1?

    yield train_index, test_index
```

or something like it maybe.
